### ISSUE TITLE
Maximum call stack size exceeded at Module.getVariableForExportName

### ISSUE THREAD

**Issue by houd1ni** [rollup/rollup#3348]
After updating core-js, rollup and this lib of mine (the commit that could break https://github.com/houd1ni/pepka/commit/d0f535c9955f4fd9a8fa76d76a8265c765f67dab), rollup-plugin-copy, mustache and typescript by just one release ahead each.

But after downgrading these, the error has not gone.

![image](https://user-images.githubusercontent.com/7501201/72728773-297b3280-3b9f-11ea-8481-f5951554c909.png)

here:

```javascript
if (name !== 'default') {
            for (const module of this.exportAllModules) {
              // console.log(name, module)
                const declaration = module.getVariableForExportName(name, true);
                if (declaration)
                    return declaration;
            }
        }
```

- Rollup Version:  1.29.0
- Operating System (or Browser): window 10 pro, latest build
- Node Version: 13.2.0

### How Do We Reproduce?

It's an NDA  project and the error is very tedious. I'll try to investigate to narrow it down, but the way isn't honestly clear. Here's a log generated by inserting console.log(module) there. Maybe it would help: https://send.firefox.com/download/ad6ee7689b9b0cb7/#vH-PmTaZafSpbPbTEs62WQ

**Comment by houd1ni** [comment]
Jesus, I found it.

What caused the problem.

I had just one orders.ts file, then it grew up and wanted a bigger nest.
I've refactored it to orders.ts with `export * from './orders'` and a new dir named `orders` with index.ts and stuff. It caused the error. Then (just now) I changed  `export * from './orders'` to  `export * from './orders/index'`. The error dissipated. Then back. Appeared again.

I hope it will help future developing, and correct this ambiguous behavior. Thanx!

**Comment by houd1ni** [comment]
Yep - the log is endless (looped) and actually been truncated. There's a sample by your code:

the last lines are looped:
```
isCordova C:/Users/pinke/Music/agg/src/utils/detect0rs.ts
isCordova C:/Users/pinke/Music/agg/src/utils/detect0rs.ts
wait C:/Users/pinke/Music/agg/src/utils/detect0rs.ts
wait C:/Users/pinke/Music/agg/src/utils/helpers.ts
lists C:/Users/pinke/Music/agg/src/api/repos/index.ts
lists C:/Users/pinke/Music/agg/src/api/repos/users.ts
lists C:/Users/pinke/Music/agg/src/api/repos/auth.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
lists C:/Users/pinke/Music/agg/src/api/repos/orders.ts
```

**Comment by houd1ni** [comment]
Actually, there're no cross-references there, btw.

**Comment by lukastaegert** [comment]
I think the error is just a "named export <xy> not found" in disguise. I will try to push a fix today.

**Comment by lukastaegert** [comment]
Fix at #3350
